src/modules/avatars/avatars.service.ts

```
@@ -0,0 +1,79 @@
+            const { url, public_id } = await this.uploadImgOrThrow(base64, type);
+            const avatar: Avatars = await this.createOne(url, public_id, transaction);
+            group.avatarId = avatar.id;
+            await group.save({transaction});
+            return group;
+        });
+    }
+    public async setAvatarToUserOrThrow(base64: string, type: string, user: Teachers | Students) {
+        return sequelize.transaction(async (transaction) => {
+            const { avatarId: oldAvatarId } = user;
+            if (oldAvatarId) {
+                await this.deleteImgFromDBaseAndCloudianry(oldAvatarId, transaction);
+            }
+            const { url, public_id } = await this.uploadImgOrThrow(base64, type);
+            const avatar: Avatars = await this.createOne(url, public_id, transaction);
+            user.avatarId = avatar.id;
+            return await user.save({transaction});
+        });
+    }
+    private async uploadImgOrThrow(base64: string, type: string) {
+        const img = `data:${type};base64,${base64}`;
+        if (!isBase64(img, {mimeRequired: true})) {
+            throw new UnsupportedMediaType('Unsupported file type');
+        }
+        try {
+            const image = await cd.uploader.upload(img, (err, res) => {
+                return err ? err : res;
+            });
+            return image;
+        } catch (e) {
+            throw new PayloadToLarge('Image size is to large');
+        }
+    }
+    private async deleteImgOrThrow(publicId: string) {
+        try {
+            const del = await cd.uploader.destroy(publicId, (err, res) => {
+                return err ? err : res;
+            });
+            return del;
+        } catch (e) {
+            throw new NotFound('Error updating avatar. Try again later');
+        }
+    }
+    private async deleteImgFromDBaseAndCloudianry(delAvatarId: number, transaction: Transaction) {
+        const oldAvatar = await this.findOneById(delAvatarId, transaction);
```

What happens when we have not found actual avatar, maybe we should throw an exception

---

An exception is not necessary, because this check is needed in order to delete the previous avatar, if it was

---

This method will be executed only if the avatar was